libs/clients/syslumenn/src/lib/syslumennClient.provider.ts

```diff
@@ -0,0 +1,40 @@
+import { Provider } from '@nestjs/common'
+
+import { createEnhancedFetch } from '@island.is/clients/middlewares'
+import { ConfigType, XRoadConfig } from '@island.is/nest/config'
+
+import { Configuration, SyslumennApi } from '../../gen/fetch'
+import { SyslumennClientConfig } from './syslumennClient.config'
+
+export const SyslumennApiProvider: Provider<SyslumennApi> = {
+  provide: SyslumennApi,
+  useFactory: (
+    config: ConfigType<typeof SyslumennClientConfig>,
+    xRoadConfig?: ConfigType<typeof XRoadConfig>,
+  ) =>
+    new SyslumennApi(
+      new Configuration({
+        fetchApi: createEnhancedFetch({
+          name: 'clients-syslumenn',
+          organizationSlug: 'syslumenn',
+          ...config.fetch,
+        }),
+        basePath:
+          xRoadConfig && config.xRoadServicePath
+            ? `${xRoadConfig.xRoadBasePath}/${config.xRoadServicePath}`
+            : config.url,
+        headers: {
+          ...(xRoadConfig &&
+            config.xRoadServicePath && {
+              'X-Road-Client': xRoadConfig.xRoadClient,
+            }),
+          'Content-Type': 'application/json',
+          Accept: 'application/json',
+        },
+      }),
+    ),
```

Comment:
This allows for misconfiguration if `config.xRoadServicesPath` is provided and `xRoadConfig` is missing the `config.url` would be used, but it's probably missing. This would fail silently until runtime execution.

So turn the arrow function into a function body and add an if statement to catch that case.

```suggestion
  ) => {
    if (config.xRoadServicePath && !xRoadConfig) {
       throw new Error('Misconfiguration in SyslumennClient. xRoadServicePath provided without xRoadConfig.')
    }
    
    return new SyslumennApi(
      new Configuration({
        fetchApi: createEnhancedFetch({
          name: 'clients-syslumenn',
          organizationSlug: 'syslumenn',
          ...config.fetch,
        }),
        basePath:
          xRoadConfig && config.xRoadServicePath
            ? `${xRoadConfig.xRoadBasePath}/${config.xRoadServicePath}`
            : config.url,
        headers: {
          ...(xRoadConfig &&
            config.xRoadServicePath && {
              'X-Road-Client': xRoadConfig.xRoadClient,
            }),
          'Content-Type': 'application/json',
          Accept: 'application/json',
        },
      }),
    ),
    }
```

Reply:
Fixed.